src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpRequestCallback.cs

```diff
@@ -149,7 +149,24 @@ private static void OnRequestReadComplete(WinHttpRequestState state, uint bytesR
             Debug.Assert(state.TcsReadFromResponseStream != null, "TcsReadFromResponseStream is null");
             Debug.Assert(!state.TcsReadFromResponseStream.Task.IsCompleted, "TcsReadFromResponseStream.Task is completed");
             
-            state.TcsReadFromResponseStream.TrySetResult((int)bytesRead);
+            state.CurrentBytesRead += (long)bytesRead;
```

Comment:
Nit: should this be moved to the else block below?  We'd only end up in the if block if bytesRead is 0, in which case the addition isn't necessary anyway.